### PR TITLE
feat: csp header

### PR DESCRIPTION
**Pull request by nugaon**
This PR research the usage of [Content Security Policy headers](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) for web3.

The problem when a dApps is loaded, it shares the [same origin](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy) with other dApps that also load from the same host (Bee client).

As the CSP's docs states: 
> If the site doesn't offer the CSP header, browsers likewise use the standard same-origin policy.

Before any same-origin policy would take effect, with CSP header it is possible to put the content into `sandbox` as at `iframe` tags. One of its consequences, the dApp will lose the origin of the P2P client from which it was served and it cannot reach other dApps from the same origin.

The extension can alter any requests/responses, therefore there is no need to configure any server-side to test the behaviours of the CSP headers.

one other consequence of this approach is the cross-domain localstorage of dapps can retrieve their origin's saved data in `iframe`s instead of the its corresponding subset of the opened root apps.

It is required from the reviewer(s) to check the concept manually with `npm run test:demo` and testing any possible security concerns regarding to the feature.

### CODE REVIEW

File: src/contentscript/document-start/index.ts [hunk 1]
```diff
@@ -1,22 +1,31 @@
...
+  dappSessionRegister(sessionId)
+  injectSessionId(sessionId)
+  injectSwarmHtml()
+  injectSwarmLibrary()
+
+  //listen to events which come from inpage side
+  new MessengerInterceptor()
+
+  // all bzz root content has to be in sandbox because of the CSP
+  if (window.origin !== 'null') return
```
Comment (Cafe137) [inline src/contentscript/document-start/index.ts:26]: Really?

File: test/bzz-protocol.spec.ts [hunk 2]
```diff
@@ -289,15 +279,34 @@ describe('BZZ protocol', () => {
...
       try {
-        await iframeWindow.window.swarm.localStorage.getItem(commonKeyName)
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        const swarm = iframeWindow.window.swarm
       } catch (e) {
-        if ((e as string).endsWith(`${commonKeyName} does not exist`)) return true
+        if (
+          e.stack &&
+          typeof e.stack === 'string' &&
+          e.stack.startsWith(`Error: Blocked a frame with origin "null" from accessing a cross-origin frame.`)
```
Comment (Cafe137) [inline test/bzz-protocol.spec.ts:288]: This is not too nice, and also seems a bit unstable.

Is there no other way of checking the error? Does it not set a `code`, `message`, `errno` or any other property?

**Review by Cafe137 — APPROVED** [review]
Hey, this LGTM - I am surprised by the `'null'` though.

File: src/contentscript/document-start/index.ts (hunk 1 above)
Comment (nugaon) [reply src/contentscript/document-start/index.ts:26]: this is the ugly truth

**Pull request merged**